### PR TITLE
Trim person spaces 3119

### PR DESCRIPTION
**Pull request by BatrakM**

### CODE REVIEW

File: apps/core/lib/core/persons/persons_api.ex [hunk 1]
```diff
@@ -12,10 +12,24 @@ defmodule Core.Persons.PersonsAPI do
 
   @person_status_active Person.status(:active)
 
+  defp trim_spaces(nil), do: nil
+  defp trim_spaces(input_string), do: input_string |> String.split() |> Enum.join(" ")
+
+  defp trim_params_spaces(params) do
+    params_to_trim = Map.take(params, ~w(first_name second_name last_name))
```
Comment (AlexKovalevych) [inline apps/core/lib/core/persons/persons_api.ex]:
```
params_to_trim = params
|> Map.take(~w(first_name second_name last_name))
|> Enum.reduce(%{}, fn {k, v}, acc -> Map.put(acc, k, trim_spaces(v)))
```

File: apps/core/lib/core/persons/persons_api.ex [hunk 2]
```diff
@@ -12,10 +12,24 @@ defmodule Core.Persons.PersonsAPI do
 
   @person_status_active Person.status(:active)
 
+  defp trim_spaces(nil), do: nil
+  defp trim_spaces(input_string), do: input_string |> String.split() |> Enum.join(" ")
+
+  defp trim_params_spaces(params) do
```
Comment (AlexKovalevych) [inline apps/core/lib/core/persons/persons_api.ex]: probably it's better to name function `trim_name_spaces`

File: apps/core/priv/repo/migrations/20181011150531_trim_spaces.exs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+defmodule Core.Repo.Migrations.TrimSpaces do
+  use Ecto.Migration
+
+  def change do
+    execute("""
+    UPDATE persons
+    SET first_name  = array_to_string(regexp_split_to_array(TRIM(first_name), ' {2,}'), ' '),
```
Comment (AlexKovalevych) [inline apps/core/priv/repo/migrations/20181011150531_trim_spaces.exs]: this migration should be fixed to work with big data (process by batches maybe)

**Comment by BatrakM** [comment]
Fixed

File: apps/core/lib/core/persons/persons_api.ex [hunk 4]
```diff
@@ -12,10 +12,20 @@ defmodule Core.Persons.PersonsAPI do
 
   @person_status_active Person.status(:active)
 
+  defp trim_spaces(nil), do: nil
+  defp trim_spaces(input_string), do: input_string |> String.split() |> Enum.join(" ")
+
+  defp trim_name_spaces(params) do
+    params
+    |> Map.take(~w(first_name second_name last_name))
+    |> Enum.reduce(%{}, fn {k, v}, acc -> Map.put(acc, k, trim_spaces(v)) end)
```
Comment (progsmile) [inline apps/core/lib/core/persons/persons_api.ex]: How about `Maybe.map(v, &trim_spaces/1)` and removing `trim_spaces` match on nil?

Comment (progsmile) [reply apps/core/lib/core/persons/persons_api.ex]: `{key, value}` will be better

File: apps/mpi/test/web/controllers/person_controller_test.exs [hunk 5]
```diff
@@ -91,6 +91,38 @@ defmodule MPI.Web.PersonControllerTest do
     assert_person(res["data"])
   end
 
+  test "spaces are trimmed when person is created", %{conn: conn} do
+    expect(KafkaMock, :publish_person_event, fn _, _, _ -> :ok end)
+
+    person_data =
+      build(:person, %{
```
Comment (progsmile) [inline apps/mpi/test/web/controllers/person_controller_test.exs]: Is there function `string_params_for` ? If not find it in eHealth, and remove Poison encode, decode

File: apps/mpi/test/web/controllers/person_controller_test.exs [hunk 6]
```diff
@@ -91,6 +91,38 @@ defmodule MPI.Web.PersonControllerTest do
...
+        first_name: "   first   name ",
+        second_name: "  second name",
+        last_name: "last name  "
+      })
+      |> Poison.encode!()
+      |> Poison.decode!()
+
+    res =
```
Comment (progsmile) [inline apps/mpi/test/web/controllers/person_controller_test.exs]: use resp, or response

File: apps/mpi/test/web/controllers/person_controller_test.exs [hunk 7]
```diff
@@ -382,6 +414,32 @@ defmodule MPI.Web.PersonControllerTest do
...
+    person = insert(:person)
+
+    person_data =
+      :person
+      |> build(%{
+        first_name: "   first   name ",
+        second_name: "  second name",
+        last_name: "last name  "
+      })
+      |> Poison.encode!()
```
Comment (progsmile) [inline apps/mpi/test/web/controllers/person_controller_test.exs]: same

**Comment by BatrakM** [comment]
Done

File: apps/core/lib/core/persons/persons_api.ex [hunk 8]
```diff
@@ -12,10 +13,19 @@ defmodule Core.Persons.PersonsAPI do
 
   @person_status_active Person.status(:active)
 
+  defp trim_spaces(input_string), do: input_string |> String.split() |> Enum.join(" ")
+
+  defp trim_name_spaces(params) do
+    params
+    |> Map.take(~w(first_name second_name last_name))
+    |> Enum.reduce(%{}, fn {key, value}, acc -> Map.put(acc, key, Maybe.map(value, &trim_spaces/1)) end)
```
Comment (AlexKovalevych) [inline apps/core/lib/core/persons/persons_api.ex]: no sense in using `reduce` here, `map` is enough

**Pull request merged**